stock_split_picking/model/stock.py

```diff
@@ -20,60 +20,47 @@
 #
 """Adds a split button on stock picking out to enable partial picking without
    passing backorder state to done"""
-from openerp import models, api, _
+from odoo import models, api, _
+from odoo.exceptions import UserError
 
 
-class stock_picking(models.Model):
+class StockPicking(models.Model):
     """Adds picking split without done state."""
 
     _inherit = "stock.picking"
 
     @api.multi
-    def split_process(self):
-        """Use to trigger the wizard from button with
-           correct context"""
-        ctx = {
-            'active_model': self._name,
-            'active_ids': self.ids,
-            'active_id': len(self.ids) and self.ids[0] or False,
-            'do_only_split': True,
-        }
+    def do_split(self):
+        reset_to_zero = self.env['stock.pack.operation']
+        for picking in self:
+            if picking.state not in ('partially_available', 'assigned'):
+                raise UserError(_("You can only split pickings "))
+            all_zeros = True
+            for operation in picking.pack_operation_ids:
+                if operation.qty_done < 0:
+                    raise UserError(_('No negative quantities allowed'))
+                if operation.qty_done > 0:
+                    operation.write({'product_qty': operation.qty_done})
+                    all_zeros = False
+            if all_zeros:
+                if picking.state != 'partially_available':
+                    raise UserError(_("You should partially collect the picking to show what amounts are to be left on this picking"))
+                else:
+                    # Since the picking is partially available, we can assume the user wants to leave the
+                    # current ungathered pack operations on this picking and split the rest off. Compare with
+                    # the normal picking logic which processes all operations if they are all at 0 gathered.
+                    for pack_op in picking.pack_operation_ids:
+                        pack_op.qty_done = pack_op.product_qty
+                    reset_to_zero += picking.pack_operation_ids
+        result = self.with_context(do_only_split=True).do_transfer()
+        # These are not backorders we've created, only splits from the original picking
```

Comment:
Allright. Of course because the core logic assumes the first part is `done` after the backorder has been created.